Tests/Unit/NetworkServer/LoRaDeviceTest.cs

```diff
@@ -876,5 +876,144 @@ public async Task When_Initialized_With_Class_C_And_Custom_RX2DR_Should_Have_Cor
 #pragma warning disable CA2000 // Dispose objects before losing scope
         private LoRaDevice CreateDefaultDevice() => new LoRaDevice("FFFFFFFF", "0000000000000000", new SingleDeviceConnectionManager(this.loRaDeviceClient.Object));
 #pragma warning restore CA2000 // Dispose objects before losing scope
+
+        public class FrameCounterInitTests
```

Comment:
Nice to see so many tests added. Just curious, why did you decide to test that logic?

Reply:
because I needed to update it with my refactoring and simplified some of the cases. I used TDD to ensure the behavior of the method wasn't change.